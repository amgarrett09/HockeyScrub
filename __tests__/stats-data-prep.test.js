/* eslint-env jest */
import DataPrep from '../api/stats-data-prep';
import mockBasicInfo from '../__mocks__/basicInfoMock';
import mockPlayerStats from '../__mocks__/playerStatsMock';

describe('prepareBasicInfo', () => {
  it('should produce an object with the correct props', () => {
    const props = DataPrep.prepareBasicInfo(mockBasicInfo);
    expect(props).toEqual({
      name: 'Tyler Bertuzzi',
      position: 'Left Wing',
      shootsCatches: 'L',
      height: '6\' 0"',
      weight: 190,
      nationality: 'CAN',
      active: true,
      currentTeam: 'Detroit Red Wings',
    });
  });

  it('should produce some default props if data does not have all info', () => {
    const emptyData = {};
    const props = DataPrep.prepareBasicInfo(emptyData);

    expect(props).toEqual({
      name: '',
      position: '',
      shootsCatches: '',
      height: '',
      weight: undefined,
      nationality: '',
      active: false,
      currentTeam: undefined,
    });
  });
});

describe('preparePlayerStats', () => {
  it('should produce correct props with good data', () => {
    const props = DataPrep.preparePlayerStats(mockPlayerStats);

    // Using a snapshot to avoid cluttering this file
    expect(props).toMatchSnapshot();
  });

  it('should produce default props if data does not exist', () => {
    const props = DataPrep.preparePlayerStats({});

    expect(props).toEqual({
      seasonStats: [],
      careerSeason: {},
      playoffStats: [],
      careerPlayoffs: {},
      lastFive: [],
    });
  });
});
