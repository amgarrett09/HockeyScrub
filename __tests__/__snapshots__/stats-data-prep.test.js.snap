// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preparePlayerStats should produce correct props with good data 1`] = `
Object {
  "careerPlayoffs": Object {},
  "careerSeason": Object {
    "assists": 43,
    "blocked": 77,
    "evenTimeOnIce": "1759:58",
    "evenTimeOnIcePerGame": "13:44",
    "faceOffPct": 30,
    "gameWinningGoals": 5,
    "games": 128,
    "goals": 28,
    "hits": 141,
    "overTimeGoals": 1,
    "penaltyMinutes": "75",
    "pim": 75,
    "plusMinus": 3,
    "points": 71,
    "powerPlayGoals": 4,
    "powerPlayPoints": 8,
    "powerPlayTimeOnIce": "194:16",
    "powerPlayTimeOnIcePerGame": "01:31",
    "shifts": 2538,
    "shortHandedGoals": 0,
    "shortHandedPoints": 0,
    "shortHandedTimeOnIce": "06:24",
    "shortHandedTimeOnIcePerGame": "00:03",
    "shotPct": 12.9,
    "shots": 217,
    "timeOnIce": "1960:38",
    "timeOnIcePerGame": "15:19",
  },
  "lastFive": Array [
    Object {
      "date": "2019-04-06",
      "game": Object {
        "content": Object {
          "link": "/api/v1/game/2018021261/content",
        },
        "gamePk": 2018021261,
        "link": "/api/v1/game/2018021261/feed/live",
      },
      "isHome": true,
      "isOT": false,
      "isWin": false,
      "opponent": Object {
        "id": 7,
        "link": "/api/v1/teams/7",
        "name": "Buffalo Sabres",
      },
      "season": "20182019",
      "stat": Object {
        "assists": 0,
        "blocked": 1,
        "evenTimeOnIce": "17:45",
        "gameWinningGoals": 0,
        "games": 1,
        "goals": 0,
        "hits": 2,
        "overTimeGoals": 0,
        "penaltyMinutes": "0",
        "pim": 0,
        "plusMinus": -1,
        "points": 0,
        "powerPlayGoals": 0,
        "powerPlayPoints": 0,
        "powerPlayTimeOnIce": "03:52",
        "shifts": 22,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "00:00",
        "shotPct": 0,
        "shots": 1,
        "timeOnIce": "21:37",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
    Object {
      "date": "2019-04-04",
      "game": Object {
        "content": Object {
          "link": "/api/v1/game/2018021245/content",
        },
        "gamePk": 2018021245,
        "link": "/api/v1/game/2018021245/feed/live",
      },
      "isHome": false,
      "isOT": false,
      "isWin": false,
      "opponent": Object {
        "id": 5,
        "link": "/api/v1/teams/5",
        "name": "Pittsburgh Penguins",
      },
      "season": "20182019",
      "stat": Object {
        "assists": 0,
        "blocked": 0,
        "evenTimeOnIce": "17:45",
        "gameWinningGoals": 0,
        "games": 1,
        "goals": 0,
        "hits": 1,
        "overTimeGoals": 0,
        "penaltyMinutes": "4",
        "pim": 4,
        "plusMinus": -1,
        "points": 0,
        "powerPlayGoals": 0,
        "powerPlayPoints": 0,
        "powerPlayTimeOnIce": "02:40",
        "shifts": 24,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "00:00",
        "shotPct": 0,
        "shots": 4,
        "timeOnIce": "20:25",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
    Object {
      "date": "2019-04-02",
      "game": Object {
        "content": Object {
          "link": "/api/v1/game/2018021233/content",
        },
        "gamePk": 2018021233,
        "link": "/api/v1/game/2018021233/feed/live",
      },
      "isHome": true,
      "isOT": false,
      "isWin": true,
      "opponent": Object {
        "id": 5,
        "link": "/api/v1/teams/5",
        "name": "Pittsburgh Penguins",
      },
      "season": "20182019",
      "stat": Object {
        "assists": 1,
        "blocked": 0,
        "evenTimeOnIce": "18:48",
        "gameWinningGoals": 1,
        "games": 1,
        "goals": 2,
        "hits": 0,
        "overTimeGoals": 0,
        "penaltyMinutes": "0",
        "pim": 0,
        "plusMinus": 2,
        "points": 3,
        "powerPlayGoals": 1,
        "powerPlayPoints": 1,
        "powerPlayTimeOnIce": "00:22",
        "shifts": 27,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "00:16",
        "shotPct": 40,
        "shots": 5,
        "timeOnIce": "19:26",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
    Object {
      "date": "2019-03-31",
      "game": Object {
        "content": Object {
          "link": "/api/v1/game/2018021219/content",
        },
        "gamePk": 2018021219,
        "link": "/api/v1/game/2018021219/feed/live",
      },
      "isHome": true,
      "isOT": false,
      "isWin": true,
      "opponent": Object {
        "id": 6,
        "link": "/api/v1/teams/6",
        "name": "Boston Bruins",
      },
      "season": "20182019",
      "stat": Object {
        "assists": 3,
        "blocked": 1,
        "evenTimeOnIce": "11:51",
        "gameWinningGoals": 0,
        "games": 1,
        "goals": 0,
        "hits": 0,
        "overTimeGoals": 0,
        "penaltyMinutes": "0",
        "pim": 0,
        "plusMinus": 1,
        "points": 3,
        "powerPlayGoals": 0,
        "powerPlayPoints": 2,
        "powerPlayTimeOnIce": "04:20",
        "shifts": 22,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "00:07",
        "shotPct": 0,
        "shots": 1,
        "timeOnIce": "16:18",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
    Object {
      "date": "2019-03-29",
      "game": Object {
        "content": Object {
          "link": "/api/v1/game/2018021199/content",
        },
        "gamePk": 2018021199,
        "link": "/api/v1/game/2018021199/feed/live",
      },
      "isHome": true,
      "isOT": false,
      "isWin": true,
      "opponent": Object {
        "id": 1,
        "link": "/api/v1/teams/1",
        "name": "New Jersey Devils",
      },
      "season": "20182019",
      "stat": Object {
        "assists": 2,
        "blocked": 0,
        "evenTimeOnIce": "15:19",
        "gameWinningGoals": 0,
        "games": 1,
        "goals": 1,
        "hits": 2,
        "overTimeGoals": 0,
        "penaltyMinutes": "0",
        "pim": 0,
        "plusMinus": 3,
        "points": 3,
        "powerPlayGoals": 0,
        "powerPlayPoints": 0,
        "powerPlayTimeOnIce": "01:00",
        "shifts": 20,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "00:00",
        "shotPct": 25,
        "shots": 4,
        "timeOnIce": "16:19",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
  ],
  "playoffStats": Array [],
  "seasonStats": Array [
    Object {
      "league": Object {
        "id": 133,
        "link": "/api/v1/league/133",
        "name": "National Hockey League",
      },
      "season": "20162017",
      "sequenceNumber": 1,
      "stat": Object {
        "assists": 0,
        "blocked": 0,
        "evenTimeOnIce": "63:23",
        "faceOffPct": 50,
        "gameWinningGoals": 0,
        "games": 7,
        "goals": 0,
        "hits": 12,
        "overTimeGoals": 0,
        "penaltyMinutes": "0",
        "pim": 0,
        "plusMinus": -1,
        "points": 0,
        "powerPlayGoals": 0,
        "powerPlayPoints": 0,
        "powerPlayTimeOnIce": "00:22",
        "shifts": 90,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "00:00",
        "shotPct": 0,
        "shots": 3,
        "timeOnIce": "63:45",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
    Object {
      "league": Object {
        "id": 133,
        "link": "/api/v1/league/133",
        "name": "National Hockey League",
      },
      "season": "20172018",
      "sequenceNumber": 1,
      "stat": Object {
        "assists": 17,
        "blocked": 38,
        "evenTimeOnIce": "617:34",
        "faceOffPct": 12.5,
        "gameWinningGoals": 1,
        "games": 48,
        "goals": 7,
        "hits": 53,
        "overTimeGoals": 0,
        "penaltyMinutes": "39",
        "pim": 39,
        "plusMinus": -7,
        "points": 24,
        "powerPlayGoals": 1,
        "powerPlayPoints": 2,
        "powerPlayTimeOnIce": "68:25",
        "shifts": 901,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "05:13",
        "shotPct": 8.4,
        "shots": 83,
        "timeOnIce": "691:12",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
    Object {
      "league": Object {
        "id": 133,
        "link": "/api/v1/league/133",
        "name": "National Hockey League",
      },
      "season": "20182019",
      "sequenceNumber": 1,
      "stat": Object {
        "assists": 26,
        "blocked": 39,
        "evenTimeOnIce": "1079:01",
        "faceOffPct": 33.33,
        "gameWinningGoals": 4,
        "games": 73,
        "goals": 21,
        "hits": 76,
        "overTimeGoals": 1,
        "penaltyMinutes": "36",
        "pim": 36,
        "plusMinus": 11,
        "points": 47,
        "powerPlayGoals": 3,
        "powerPlayPoints": 6,
        "powerPlayTimeOnIce": "125:29",
        "shifts": 1547,
        "shortHandedGoals": 0,
        "shortHandedPoints": 0,
        "shortHandedTimeOnIce": "01:11",
        "shotPct": 16,
        "shots": 131,
        "timeOnIce": "1205:41",
      },
      "team": Object {
        "id": 17,
        "link": "/api/v1/teams/17",
        "name": "Detroit Red Wings",
      },
    },
  ],
}
`;
